lib/bitcoin/BitcoinClient.ts

```diff
@@ -68,20 +69,66 @@ export default class BitcoinClient {
    * Initialize this bitcoin client.
    */
   public async initialize (): Promise<void> {
+    // Periodically poll Bitcoin Core status until it is ready.
+    const bitcoinCoreStatusPollingWindowInSeconds = 60;
+    await this.waitUntilBitcoinCoreIsReady(bitcoinCoreStatusPollingWindowInSeconds);
+
+    await this.initializeBitcoinCore();
+  }
+
+  /**
+   * Periodically polls Bitcoin Core status until it is ready.
+   * @param pollingWindowInSeconds Time to wait between each status check. Mainly used for speeding up unit tests.
+   */
+  private async waitUntilBitcoinCoreIsReady (pollingWindowInSeconds: number): Promise<void> {
+    while (true) {
+      try {
+        Logger.info('Getting blockchain info...');
+        const request = {
+          method: 'getblockchaininfo'
+        };
+
+        const isWalletRpc = false;
+        const allowTimeout = true;
+        const response = await this.rpcCall(request, allowTimeout, isWalletRpc);
+        const blockHeight = response.headers;
+        const syncedBlockHeight = response.blocks;
+
+        Logger.info(LogColor.lightBlue(`Bitcoin sync progress: block height ${LogColor.green(blockHeight)}, sync-ed: ${LogColor.green(syncedBlockHeight)}`));
+
+        if (syncedBlockHeight === blockHeight) {

```

Comment:
Could this check ever be true if block headers and block height have not been synced? So Height and Headers are both zero?

Not sure if this could be the case but maybe make sure headers are not zero?

Reply:
Good call! I felt it was edge case as syncing header is literally the first thing bitcoin core does when it fires up, but better safe than sorry, 'coz if network is down upon start up or something...